stat/distmat/permutation_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distmat
+
+import (
+	"testing"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/mat"
+)
+
+func TestUniformPermutation(t *testing.T) {
+	up := NewUniformPermutation(rand.NewSource(1))
+	for _, n := range []int{10, 32, 64, 100} {
+		m := mat.NewDense(n, n, nil)
+		if m == nil {
+			t.Error("Matrix failed")
+		}
+		up.PermTo(m)
+		r, c := m.Dims()
+		if r != n || c != n {
+			t.Error("got back matrix of wrong size")
+		}
+		confirmPermMatrix(t, m)
+	}
+
+}
```

Comment:
Please use the test code in https://github.com/gonum/gonum/pull/1417#discussion_r466353985